libraries/lib-graphics/graphics/Point.h

```diff
@@ -0,0 +1,192 @@
+   {
+      x = static_cast<DataType>(x / scale);
+      y = static_cast<DataType>(y / scale);
+
+      return *this;
+   }
+
+   PointType operator-() const noexcept
+   {
+      static_assert(std::is_signed_v<DataType>);
+
+      return { -x, -y };
+   }
+
+   bool IsZero() const noexcept
+   {
+      return std::abs(x) <= std::numeric_limits<DataType>::epsilon() &&
+             std::abs(y) <= std::numeric_limits<DataType>::epsilon();
+   }
+};
+
+//! Casts a point to another point type.
+template <typename To, typename From>
+PointType<To> point_cast(PointType<From> point)
+{
+   return { static_cast<To>(point.x), static_cast<To>(point.y) };
+}
+
+template <typename DataType>
+bool operator==(PointType<DataType> lhs, PointType<DataType> rhs) noexcept
+{
+   return lhs.x == rhs.x && lhs.y == rhs.y;
+}
+
+template <typename DataType>
+bool operator!=(PointType<DataType> lhs, PointType<DataType> rhs) noexcept
+{
+   return !(lhs == rhs);
+}
+
+template <typename DataType>
+PointType<DataType>
+operator+(PointType<DataType> lhs, PointType<DataType> rhs) noexcept
+{
+   return { lhs.x + rhs.x, lhs.y + rhs.y };
+}
+
+template <typename DataType>
+PointType<DataType>
+operator-(PointType<DataType> lhs, PointType<DataType> rhs) noexcept
+{
+   return { lhs.x - rhs.x, lhs.y - rhs.y };
+}
+
+template <typename DataType>
+PointType<DataType>
+operator*(PointType<DataType> lhs, PointType<DataType> rhs) noexcept
+{
+   return { lhs.x * rhs.x, lhs.y * rhs.y };
+}
+
+template <typename DataType>
+PointType<DataType>
+operator/(PointType<DataType> lhs, PointType<DataType> rhs) noexcept
+{
+   return { lhs.x / rhs.x, lhs.y / rhs.y };
+}
+
+template <typename DataType, typename ScaleType>
+PointType<DataType> operator*(PointType<DataType> lhs, ScaleType rhs) noexcept
+{
+   return { static_cast<DataType>(lhs.x * rhs),
+            static_cast<DataType>(lhs.y * rhs) };
+}
+
+template <typename DataType, typename ScaleType>
+PointType<DataType> operator*(ScaleType lhs, PointType<DataType> rhs) noexcept
+{
+   return { static_cast<DataType>(lhs * rhs.x),
+            static_cast<DataType>(lhs * rhs.y) };
+}
+
+template <typename DataType, typename ScaleType>
+PointType<DataType> operator/(PointType<DataType> lhs, ScaleType rhs) noexcept
+{
+   return { static_cast<DataType>(lhs.x / rhs),
+            static_cast<DataType>(lhs.y / rhs) };
+}
+
+//! Returns the dot (inner) product between the vectors, defined by points.
+template <typename DataType>
+auto DotProduct(PointType<DataType> lhs, PointType<DataType> rhs) noexcept
+{
+   return lhs.x * rhs.x + lhs.y * rhs.y;
+}
+
+//! Returns the length of a vector matching the point.
+template <typename DataType>
+auto Norm(PointType<DataType> lhs) noexcept
+{
+   return std::sqrt(DotProduct(lhs, lhs));
+}
+
+//! Returns the distance between the two points
+template <typename DataType>
+auto Distance(PointType<DataType> lhs, PointType<DataType> rhs) noexcept
+{
+   return std::sqrt(DotProduct(lhs, rhs));
+}
```

Comment:
That's wrong.  Distance between points is norm of the difference.

Reply:
This one is an unfortunate mistake